File: ios/Checkout.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(Checkout, NSObject)

RCT_EXTERN_METHOD(initiatePayment:(NSString)publicKey
                  withEnvironment:(NSString)environment
                 withResolver:(RCTPromiseResolveBlock)resolve
                 withRejecter:(RCTPromiseRejectBlock)reject)

@end

File: ios/Checkout.swift
@objc(Checkout)
class Checkout: NSObject {
    let vc: CheckoutViewController = CheckoutViewController.init();
    @objc(initiatePayment:withEnvironment:withResolver:withRejecter:)
    func initiatePayment(publicKey: String, environment: String,
                         resolve:RCTPromiseResolveBlock,reject:RCTPromiseRejectBlock) -> Void {
        
        do
        {
            DispatchQueue.main.async {
                self.vc.initiatePayment(publicKey: publicKey, environment: environment);
            }
            resolve(true);
        } catch {
            // Add dispach_queue
            DispatchQueue.main.async {
              print(error)
            }
            resolve(false);
        }
    }
}
